workshops/models.py

```diff
@@ -227,6 +227,16 @@ def unpublished_events(self):
 
         return self.filter(published=False)
 
+    def unpaid_events(self):
+        '''Return a queryset for events that owe money.'''
+
+        # All events that have a non-zero admin fee.
+        queryset = self.filter(admin_fee__ne=0)
+
+        # Of those, all those whose paid status is not 'Yes' (i.e., is 'No' or NULL).
+        queryset = queryset.filter(fee_paid__ne=True)
```

Comment:
On Wed, May 06, 2015 at 01:14:18PM -0700, Piotr Banaszkiewicz wrote:
> Not sure if it's possible to store negative numbers in DecimalField,
> but I thought a little and it makes no sense to list overpaid events
> under "unpaid" label.

Hrm, “overpaid” makes it sound like “we're adjusting admin_fee to be
the outstanding balance for this event”.  I'd expected it to be “how
much we charged (or will charge) for this event”, with payment status
only showing up via the fee_paid boolean.  If we *do* want to track
received money, I'd rather make fee_paid a DecimalField (or add a
separate Payment table tracking per-host payment instances).  But
adjusting admin_fee after the fact is going to make it hard to answer
questions like “how much did we charge for their last workshop?” or
even “how much money should we have earned last month?”.